temporalcli/commands.workflow_test.go

```diff
@@ -407,51 +427,108 @@ func (s *SharedServerSuite) TestWorkflow_Update() {
-		"--name", updateName, "-i", strconv.Itoa(input))
-	s.NoError(res.Err)
-	s.Contains(res.Stdout.String(), strconv.Itoa(input))
+	// successful update, output should contain the result
+	res := t.execute("workflow", "update", "execute", "--address", t.s.Address(), "-w", run.GetID(), "--name", updateName, "-i", strconv.Itoa(input))
+	t.s.NoError(res.Err)
+	t.s.ContainsOnSameLine(res.Stdout.String(), "Result", strconv.Itoa(1*input))
 
 	// successful update passing first-execution-run-id
-	res = s.Execute("workflow", "update", "--address", s.Address(), "-w", run.GetID(),
-		// Use --type here to make sure the alias works
-		"--type", updateName, "-i", strconv.Itoa(input), "--first-execution-run-id", run.GetRunID())
-	s.NoError(res.Err)
+	// Use --type here to make sure the alias works
+	res = t.execute("workflow", "update", "execute", "--address", t.s.Address(), "-w", run.GetID(), "--type", updateName, "-i", strconv.Itoa(input), "--first-execution-run-id", run.GetRunID())
+	t.s.NoError(res.Err)
+	t.s.ContainsOnSameLine(res.Stdout.String(), "Result", strconv.Itoa(2*input))
 
 	// successful update passing update-id
-	res = s.Execute("workflow", "update", "--address", s.Address(), "--update-id", strconv.Itoa(input), "-w", run.GetID(), "--name", updateName, "-i", strconv.Itoa(input))
-	s.NoError(res.Err)
-	s.Contains(res.Stdout.String(), strconv.Itoa(input))
-	res = s.Execute("workflow", "update", "--address", s.Address(), "--update-id", strconv.Itoa(input), "-w", run.GetID(), "--name", updateName)
-	s.NoError(res.Err)
-	s.Contains(res.Stdout.String(), strconv.Itoa(input))
+	res = t.execute("workflow", "update", "execute", "--address", t.s.Address(), "--update-id", strconv.Itoa(input), "-w", run.GetID(), "--name", updateName, "-i", strconv.Itoa(input))
+	t.s.NoError(res.Err)
+	t.s.ContainsOnSameLine(res.Stdout.String(), "UpdateID", strconv.Itoa(input))
+	t.s.ContainsOnSameLine(res.Stdout.String(), "Result", strconv.Itoa(3*input))
+
+	// successful update without input
+	res = t.execute("workflow", "update", "execute", "--address", t.s.Address(), "--update-id", strconv.Itoa(input), "-w", run.GetID(), "--name", updateName)
+	t.s.NoError(res.Err)
+	t.s.ContainsOnSameLine(res.Stdout.String(), "UpdateID", strconv.Itoa(input))
+	t.s.ContainsOnSameLine(res.Stdout.String(), "Result", strconv.Itoa(3*input))
+
+	if t.useStart {
+		// update rejected, name not supplied
+		res = t.s.Execute("workflow", "update", "start", "--wait-for-stage", "accepted", "--address", t.s.Address(), "-w", run.GetID(), "-i", strconv.Itoa(input))
+		t.s.ErrorContains(res.Err, "required flag(s) \"name\" not set")
+
+		// update rejected, wrong workflowID
+		res = t.s.Execute("workflow", "update", "start", "--wait-for-stage", "accepted", "--address", t.s.Address(), "-w", "nonexistent-wf-id", "--name", updateName, "-i", strconv.Itoa(input))
+		t.s.ErrorContains(res.Err, "unable to update workflow")
+
+		// TODO: wrong update name is not currently an error when using `update start` (?!)
```

Comment:
- [ ] TODO (blocking): investigate

Reply:
Fixed in latest commits; thanks @Quinn-With-Two-Ns and @cretz for discussion there.